boards/DCM/Inc/Io/Io_LSM6DS33.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+/**
+ * Driver for interfacing with the IMU LSM6DS33 via I2c
+ *
+ * The Application Note for this IMU can be found here:
+ * https://www.pololu.com/file/0J1088/LSM6DS33-AN4682.pdf
+ */
+
+#include <stdint.h>
+#include <stm32f3xx_hal.h>
+#include <stdbool.h>
+#include <App_SharedExitCode.h>
+
+struct ImuData;
+
+/**
+ * Configures the LSM6DS33 IMU Gyro/Accelerometer chip
+ * @param i2c_handle The handle for the I2C device the IMU is connected to
+ * @return None
+ */
+HAL_StatusTypeDef Io_LSM6DS33_ConfigureImu(I2C_HandleTypeDef *i2c_handle);
+
+/**
+ * Read data from the IMU and update internal variables
+ * @return ExitCode which is EXIT_CODE_OK if successful, EXIT_CODE_OUT_OF_RANGE
+ * if data is not within specified range, EXIT_CODE_INVALID_ARGS if data not
+ * read successfully
+ */
+ExitCode Io_LSM6DS33_UpdateSensorData(void);
```

Comment:
Looked into the IO code and read up on the documentation a little bit. This function should be called Io_LSM6DS33_ExtiCallback. 

The IMU is currently generating an interrupt on the INT2 pin when the acceleration/angular velocity values are updated. We want to call this function inside the callback function that is triggered when the MCU detects the external interrupt.

Reply:
ok, will do this after InRangeCheck stuff

Reply:
done